Libraries/Hotcakes.Commerce.Dnn/Web/HotcakesModuleBase.cs

```diff
@@ -145,25 +145,48 @@ protected override void OnPreRender(EventArgs e)
 
         protected void RegisterStyleSheet(string path)
         {
+
             ClientResourceManager.RegisterStyleSheet(Page, ResolveUrl(path), FileOrder.Css.ModuleCss);
         }
 
         protected void RegisterViewScript(string path, int order = 0)
         {
-            ClientResourceManager.RegisterScript(Page, ResolveUrl(HccApp.ViewsVirtualPath + "/Scripts/" + path),
-                FileOrder.Js.DefaultPriority + 10 + order);
+            if (!string.IsNullOrEmpty(path))
+            {
+                ClientResourceManager.RegisterScript(Page, ResolveUrl(HccApp.ViewsVirtualPath + "/Scripts/" + path),
+                    FileOrder.Js.DefaultPriority + 10 + order);
+            }
+            else
+            {
+                throw new ArgumentNullException(nameof(path));
+            }
         }
 
         protected void RegisterScript(string path, int order = 0)
         {
-            ClientResourceManager.RegisterScript(Page, ResolveUrl(ScriptsBasePath + path),
-                FileOrder.Js.DefaultPriority + order);
+            if (!string.IsNullOrEmpty(path))
```

Comment:
This will represent a change in functionality.  Do you see a scenario where a developer might be okay with and depend on the non-exception approach?  What ends up in the HTML markup before this update?

Reply:
This method adds the name of the js file to the base URL to build the full path of the file. If the file name is null the URL would be incomplete and as a result, in the HTML view, this script returns a 404 when loaded. On the other hand, if we handle the null exception for the path field the view would show an error. I consider continuing to use this method as it was